FrameworkSystem/Client/AuthManagerData.py

```diff
@@ -0,0 +1,261 @@
+  # #         ProxyProvider: [ <proxy providers> ],
+  # #         VOMSRoles: [ <VOMSRoles> ],
+  # #         ...
+  # #       },
+  # #       <DN2>: { ... },
+  # #     }
+  # #   },
+  # #   <ID2>: { ... }
+  # # }
+
+  __service = DictCache()
+  # # {
+  # #   crash: bool
+  # # }
+
+  __metaclass__ = DIRACSingleton.DIRACSingleton
+
+  @gCacheProfiles
+  def getProfiles(self, userID=None):
+    """ Get cache information
+
+        :param str userID: user ID
+
+        :return: dict
+    """
+    if userID:
+      return self.__cacheProfiles.get(userID) or {}
+    return self.__cacheProfiles.getDict()
+
+  @gCacheProfiles
+  def updateProfiles(self, data, time=3600 * 24):
+    """ Add cache information
+
+        :param dict data: ID information data
+        :param int time: lifetime
+    """
+    for oid, info in data.items():
+      self.__cacheProfiles.add(oid, time, value=info)
+
+  @gCacheSessions
+  def getSessions(self, session=None, userID=None):
+    """ Get cache information
+
+        :param str session: session
+        :param str userID: user ID
+
+        :return: dict
+    """
+    if session:
+      return self.__cacheSessions.get(session) or {}
+    if userID:
+      sessions = {}
+      for session in self.__cahceIDs.get(userID) or []:
+        sDict = self.__cacheSessions.get(session)
+        if sDict:
+          sessions[session] = sDict
+      return sessions
+    return self.__cacheSessions.getDict()
+
+  @gCacheSessions
+  def updateSessions(self, data, time=3600 * 24):
+    """ Get cache information
+
+        :param dict data: session information data
+        :param int time: lifetime
+    """
+    for session, info in data.items():
+      idSessions = self.__cahceIDs.get(info['ID']) or []
+      self.__cahceIDs.add(info['ID'], time, list(set(idSessions + [session])))
+      self.__cacheSessions.add(session, time, value=info)
+
+  def resfreshSessions(self, session=None, userID=None):
+    """ Refresh session cache from service
+
+        :param str session: session to update
+        :param str userID: user ID
+
+        :return: S_OK()/S_ERROR()
+    """
+    servCrash = self.__service.get('crash')
+    if servCrash:
+      return servCrash
+    from DIRAC.Core.DISET.RPCClient import RPCClient
+    result = RPCClient('Framework/AuthManager', timeout=10).getSessionsInfo(session, userID)
+    # If the AuthManager service is down client will ignore it 1 minute
+    if result.get('Errno', 0) == 1112:
+      self.__service.add('crash', 60, value=result)
+    if result['OK'] and result['Value']:
+      self.updateSessions({session: result['Value']} if session else result['Value'])
+    return result
+
+  def resfreshProfiles(self, userID=None):
+    """ Refresh profiles cache from service
+
+        :param str userID: userID to update
+
+        :return: S_OK()/S_ERROR()
+    """
+    servCrash = self.__service.get('crash')
+    if servCrash:
+      return servCrash
+    from DIRAC.Core.DISET.RPCClient import RPCClient
```

Comment:
Please don't call `RPCClient` directly. Use the `Client`

Reply:
Used `Client` now